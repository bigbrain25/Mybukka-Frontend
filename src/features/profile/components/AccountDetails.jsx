import React from 'react';

import InputAccountDetails from '../common/InputAccountDetails';
import AccountDetailsGroupHeader from '../common/AccountDetailsGroupHeader';
import Addresses from './Addresses';
import Payment from './Payment';

import './accountDetails.scss';

const AccountDetails = () => (
  <div className="account-details">
    <AccountDetailsGroupHeader text="Account Details" />
    <InputAccountDetails placeHolder="First name" name="firstName" />
    <InputAccountDetails placeHolder="Last name" name="lastName" />
    <InputAccountDetails placeHolder="Email" name="email" type="email" />
    <InputAccountDetails
      placeHolder="Contact Mobile"
      name="contactMobile"
      type="number"
    />
    <Addresses addresses={['1, Aminu street, Mende']} />
    <Payment />
  </div>
);

export default AccountDetails;
