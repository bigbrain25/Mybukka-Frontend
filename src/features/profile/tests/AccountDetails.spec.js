import React from 'react';
import { Provider } from 'react-redux';
import AccountDetails from '../components/AccountDetails';

const store = mockStore({
  deliveryModeReducer: {
    mode: 'pickup'
  },
  loadingReducer: { status: false },
  authenticationReducer: { status: { authenticated: true }
  },
  fetchUserData: { userInfo: {
    firstName: 'save',
    lastName: 'save',
    email: 'save',
    contactMobile: 'save'
  },
  finishedRequest: true },
  fetchUserAddress: { userAddresses: [
    { address: '15, onayade street' }
  ] },
  postUserData: { errorMessage: '' },
  postUserAddress: { errorMessage: '', posted: false },
  selectedLocationReducer: { coordinates: [3.361476, 6.5560715] }
});

describe('Account Details components', () => {
  const props = {
    requestUserAddress: jest.fn(),
    deleteUserAddress: jest.fn(),
    requestUserData: jest.fn(),
    loading: false,
    userInfo: {
      firstName: 'save',
      lastName: 'save',
      email: 'save',
      contactMobile: 'save'
    },
    userAddress: [{ address: 'address' }],
    editUserData: jest.fn()
  };
  const wrapper = mount(
    <Provider store={store}>
      <AccountDetails {...props} />
    </Provider>
  );

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
