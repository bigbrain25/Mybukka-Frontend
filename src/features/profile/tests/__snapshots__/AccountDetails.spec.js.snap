// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Details components renders properly 1`] = `ReactWrapper {}`;
